.github/workflows/example.yml

```diff
@@ -0,0 +1,33 @@
+name: Run example
+on:
+  workflow_dispatch:
+    inputs:
+      example_paths:
+        type: string
+        required: true
+  workflow_call:
+    inputs:
+      example_paths:
+        type: string
+        required: true
+jobs:
+  run_example:
+    name: Run example
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        example_path: ${{ fromJSON(inputs.example_paths) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: denoland/setup-deno@004814556e37c54a2f6e31384c9e18e983317366 # v1.1.0
+        with:
+          deno-version: v1.x
+      - name: Cache Deno dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/deno
+          key: ${{ runner.os }}-build-${{ hashFiles('lock.json', 'deps/**/*.ts') }}
```

Comment:
added `deps/*` to the hash because `deps/*` it's not always in sync with the `lock.json` file.